pkg/kritis/util/util.go

```diff
@@ -62,12 +63,37 @@ func GetResource(image string) *grafeas.Resource {
 	return &grafeas.Resource{Uri: GetResourceURL(image)}
 }
 
-func GetPgpAttestationFromOccurrence(occ *grafeas.Occurrence) metadata.PGPAttestation {
-	pgp := occ.GetAttestation().GetAttestation().GetPgpSignedAttestation()
-	return metadata.PGPAttestation{
-		Signature: pgp.GetSignature(),
-		KeyID:     pgp.GetPgpKeyId(),
-		OccID:     occ.GetName(),
+func GetRawAttestationFromOccurrence(occ *grafeas.Occurrence) *metadata.RawAttestation {
+	signatureType := metadata.UnknownSignatureType
+	var signatures []metadata.RawSignature
+	var serializedPayload []byte
+
+	att := occ.GetAttestation().GetAttestation()
+	switch att.Signature.(type) {
+	case *attestationpb.Attestation_PgpSignedAttestation:
+		psa := att.GetPgpSignedAttestation()
+		signatureType = metadata.PgpSignatureType
+		sig := metadata.RawSignature{
+			PublicKeyId: psa.GetPgpKeyId(),
+			Signature:   psa.GetSignature(),
+		}
+		signatures = append(signatures, sig)
+	case *attestationpb.Attestation_GenericSignedAttestation:
+		gsa := att.GetGenericSignedAttestation()
+		signatureType = metadata.GenericSignatureType
+		serializedPayload = gsa.GetSerializedPayload()
+		for _, sig := range gsa.GetSignatures() {
+			newSig := metadata.RawSignature{
+				PublicKeyId: sig.PublicKeyId,
+				Signature:   string(sig.Signature),
+			}
+			signatures = append(signatures, newSig)
+		}
+	}
```

Comment:
Throw error for unknown signature type?

Reply:
Done. Should I add a test case for this as well?